homeassistant/components/ipp/__init__.py

```diff
@@ -41,29 +41,33 @@
 
 async def async_setup(hass: HomeAssistant, config: Dict) -> bool:
     """Set up the IPP component."""
-    hass.data.setdefault(DOMAIN, {})
+    if DOMAIN not in hass.data:
+        hass.data.setdefault(DOMAIN, {})
     return True
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up IPP from a config entry."""
 
-    # Create IPP instance for this entry
-    coordinator = IPPDataUpdateCoordinator(
-        hass,
-        host=entry.data[CONF_HOST],
-        port=entry.data[CONF_PORT],
-        base_path=entry.data[CONF_BASE_PATH],
-        tls=entry.data[CONF_SSL],
-        verify_ssl=entry.data[CONF_VERIFY_SSL],
-    )
+    if entry.entry_id not in hass.data[DOMAIN]:
```

Comment:
Alternatively:

```py
coordinator = hass.data[DOMAIN].get(entry.entry_id)

if not coordinator:
  coordinator = IPPDataUpdateCoordinator(
  ...
  hass.data[DOMAIN][entry.entry_id] = coordinator
  
await coordinator.async_refresh()
```

Not really needed and personal preference a bit... As I generally try to avoid if/else constructs :)

Reply:
looks much better, thank you!

this is the sort of code you get when you program at 1h30 AM 😴